lib/matplotlib/axes/_base.py

```diff
@@ -985,14 +1040,30 @@ def clear(self):
         """clear the axes"""
         self.cla()
 
+    def set_prop_cycle(self, prop_cycle):
+        """
+        Set the prop cycle for any future plot commands on this Axes.
+
+        *prop_cycle* is a :class:Cycler object.
+        Can also be `None` to reset to the cycle defined by the
+        current style.
+        """
+        self._get_lines.set_prop_cycle(prop_cycle)
+        self._get_patches_for_fill.set_prop_cycle(prop_cycle)
+
     def set_color_cycle(self, clist):
         """
         Set the color cycle for any future plot commands on this Axes.
 
         *clist* is a list of mpl color specifiers.
+
+        .. deprecated:: 1.5
```

Comment:
sure, but take this function to its logical conclusion, we have a ``set_color_cycle`` function (and we want to keep it), then surely logically speaking we should also have similar functions for other properties...

```python
import matplotlib.pyplot as plt
fig, ax = plt.subplot()
ax.set_color_cycle(['r', 'g', 'b'])
ax.set_lw_cycle([1, 2, 3])
```

Here I would expect the two to work independently from each other, i.e. an inner product of the two cycles.  I wouldn't expect the second to change the first... if it were a method of ``set_property_cycle_as_color_cycle``, then yes, but like this no.

I appreciate that it looks difficult, but I feel it warrants more investigation, it feels possible, but perhaps I just listen to my inner engineer...

> Normal people believe that if it ain't broke, don't fix it.
>
> Engineers believe that if it ain't broke, it doesn't have enough features yet.
> -- Scott Adams